test/trento/users_test.exs

```diff
@@ -117,6 +117,17 @@ defmodule Trento.UsersTest do
       assert length(users) == 1
     end
 
+    test "list_all_usernames returns all usernames including those for deleted users" do
+      %{username: username1} = insert(:user)
+      %{username: username2} = insert(:user, deleted_at: DateTime.utc_now())
+      inserted_sorted_usernames = Enum.sort([username1, username2])
```

Comment:
The sorting here in tests is only to make the equality assertion below on  the list of usernames deterministic/non-flaky.

Reply:
That makes totally sense, well done! 
I was just wondering whether `Users.list_all_usernames` already does some sorting (it does not look it does) and if it makes sense adding alphabetical sorting to the query.

What changes is that:
- `Users.list_all_usernames` would deterministically return an alphabetically sorted list of usernames
- in the test we only have to sort `inserted_sorted_usernames`

Reply:
After sync conversation, ordering in-query that fetches the usernames list doesn't help much since we also add in the system user to the list of users in the channel code.